pkg/testing/tools/artifacts_api.go

```diff
@@ -127,15 +134,17 @@ func WithHttpClient(client httpDoer) ArtifactAPIClientOpt {
 // More information about the API can be found at https://artifacts-api.elastic.co/v1
 // which will print a list of available operations
 type ArtifactAPIClient struct {
-	c   httpDoer
-	url string
+	c      httpDoer
+	url    string
+	cdnURL string
 }
 
 // NewArtifactAPIClient creates a new Artifact API client
 func NewArtifactAPIClient(opts ...ArtifactAPIClientOpt) *ArtifactAPIClient {
 	c := &ArtifactAPIClient{
-		url: defaultArtifactAPIURL,
-		c:   new(http.Client),
+		url:    defaultArtifactAPIURL,
+		cdnURL: artifactReleaseCDN,
+		c:      new(http.Client),
```

Comment:
not your PR but there's a potential eternal hang

Reply:
it's not obvious to me, can you elaborate more?

Reply:
Go default http client does not have a timeout

Reply:
@AndersonQ this code is used only for testing, would not our test timeout stop the test anyway?

Reply:
They will, besides using a context when creating the requests, as you did, will respect the context's cancellation and interrupt the request.

Reply:
anyway, for me it isn't a blocker and I'd say it's resolved.
@michalpristas, is it solved for you?

Reply:
sorry for late response, this was not a blocker for me